### PR TITLE
fix(dashboard): move header to top

### LINKED ISSUE #32104: Dependency Dashboard header no longer always on top
### Discussed in https://github.com/renovatebot/renovate/discussions/32103

<div type='discussions-op-text'>

<sup>Originally posted by **astellingwerf** October 23, 2024</sup>
### How are you running Renovate?

None

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

_No response_

### Please tell us more about your question or problem

According to documentation, the `dependencyDashboardHeader` is "Any text added here will be placed first in the Dependency Dashboard issue body."`. This is not true anymore, since #31129 was added. The header is now under the H2 for "Config(uration?) Migration Needed":

```
## Config Migration Needed

<!-- config-migration-pr-info --> See Config Migration MR: #93.

This issue lists Renovate updates and detected dependencies. Read the [Dependency Dashboard](https://docs.renovatebot.com/key-concepts/dashboard/) docs to learn more.

> ⚠ **Warning**
> 
These dependencies are deprecated:

| Datasource | Name | Replacement MR? |
|------------|------|--------------|
| dockerfile | `node` | ![Available](https://img.shields.io/badge/available-green?style=flat-square) |
```

In my opinion, the section that @RahulGautamSingh added to `ensureDependencyDashboard` for it should be shifted down to after the header.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>
</div>

### PR DESCRIPTION
**Pull request by tasneemkoushar**
## Changes
Refactored code to bring Dependency Dashboard header on top

## Context

Closes #32104

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/dependency-dashboard.ts [hunk 1]
```diff
@@ -286,6 +286,11 @@ export async function ensureDependencyDashboard(
   }
   let issueBody = '';
 
+  if (config.dependencyDashboardHeader?.length) {
```
Comment (HonkingGoose) [inline lib/workers/repository/dependency-dashboard.ts:289]: I guess your PR will fix this issue?

- #32104

If you're closing that issue, please add this text to your PR's body text (the top post):

`Closes #32104`

This way GitHub automatically closes the linked issue when your PR gets merged.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
I think we need to link to the issue this PR closes, before we merge it?

File: lib/workers/repository/dependency-dashboard.ts (hunk 1 above)
Comment (rarkins) [reply lib/workers/repository/dependency-dashboard.ts:289]: I added it

**Review by rarkins — APPROVED** [review]

**Review by HonkingGoose — APPROVED** [review]
The PR now properly closes the related issue when it gets merged. I'll let the maintainers review the code.

**Review by RahulGautamSingh — APPROVED** [review]

**Pull request merged**